internal/batches/executor/execution_cache.go

```diff
@@ -166,18 +236,47 @@ func (c ExecutionDiskCache) Clear(ctx context.Context, key ExecutionCacheKey) er
 	return os.Remove(path)
 }
 
-// ExecutionNoOpCache is an implementation of actionExecutionCache that does not store or
+func (c ExecutionDiskCache) GetStepResult(ctx context.Context, key CacheKeyer) (stepExecutionResult, bool, error) {
+	var result stepExecutionResult
+	path, err := c.cacheFilePath(key)
+	if err != nil {
+		return result, false, err
+	}
+
+	found, err := c.readCacheFile(path, &result)
+
+	return result, found, nil
+}
+
+func (c ExecutionDiskCache) SetStepResult(ctx context.Context, key CacheKeyer, result stepExecutionResult) error {
+	path, err := c.cacheFilePath(key)
+	if err != nil {
+		return err
+	}
+
+	return c.writeCacheFile(path, &result)
+}
+
+// ExecutionNoOpCache is an implementation of ExecutionCache that does not store or
 // retrieve cache entries.
 type ExecutionNoOpCache struct{}
```

Comment:
I saw we call the log manager `NoopLogManager`, would it make sense to rename this `ExecutionNoopCache` for consistency?

Reply:
Ah, good call. But I'll do the opposite and change `NoopLogManager` to `LogNoOpManager` since `ExecutionNoOpCache` is older and I don't see a reason for changing the old pattern to the new one.